src/wallet/rpcwallet.cpp

```diff
@@ -3463,11 +3463,44 @@ UniValue z_listreceivedbyaddress(const UniValue& params, bool fHelp)
     if (nMinDepth < 0) {
         throw JSONRPCError(RPC_INVALID_PARAMETER, "Minimum number of confirmations cannot be less than 0");
     }
+    UniValue result(UniValue::VARR);
 
     // Check that the from address is valid.
     auto fromaddress = params[0].get_str();
 
+    auto push_transparent_result = [&](const CTxDestination& dest) -> void {
+        const CScript scriptPubKey{GetScriptForDestination(dest)};
+        for (const auto& [_txid, wtx] : pwalletMain->mapWallet) {
+            if (wtx.IsCoinBase() || !CheckFinalTx(wtx))
+                continue;
+
+            int nDepth = wtx.GetDepthInMainChain();
+            if (nDepth < nMinDepth) continue;
+            for (size_t i = 0; i < wtx.vout.size(); ++i) {
+                const CTxOut& txout{wtx.vout[i]};
+                if (txout.scriptPubKey == scriptPubKey) {
+                    UniValue obj(UniValue::VOBJ);
+                    obj.pushKV("pool", "transparent");
+                    obj.pushKV("txid", wtx.GetHash().ToString());
+                    obj.pushKV("amount", ValueFromAmount(txout.nValue));
+                    obj.pushKV("amountZat", txout.nValue);
+                    obj.pushKV("outindex", int(i));
+                    obj.pushKV("confirmations", nDepth);
+                    result.push_back(obj);
+                }
+            }
+        }
+    };
     KeyIO keyIO(Params());
+
+    // Is the address argument a transparent address?
+    CTxDestination dest = keyIO.DecodeDestination(fromaddress);
+    if (IsValidDestination(dest)) {
+        push_transparent_result(dest);
+        return result;
+    }
+
```

Comment:
```suggestion
```
We should not use `DecodeDestination` here; instead, just call `push_transparent_result` from within the handlers for transparent variants of `decoded`. You've already factored this out as a lambda; just invoke it in the `decoded` handlers. My eventual objective is to remove `keyIO.DecodeDestination` and `CTxDestination` entirely.